.github/workflows/ci.yml

```
@@ -1,15 +1,43 @@
 name: CI
-on:
-  push: {}
-  pull_request: {}
+on: [push, pull_request]
 
 jobs:
   run:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - run: corepack enable
       - run: pnpm install
       - run: pnpm run build
       - run: pnpm run lint
       - run: pnpm test
+
+  test-types:
+    runs-on: ubuntu-latest
+    name: Test TypeScript Types with Node.js ${{ matrix.node }} and TypeScript ${{ matrix.ts }}
+    strategy:
+      matrix:
+        node: [20.x]
+        ts: [4.7, 4.8, 4.9, 5.0, 5.1, 5.2, 5.3, 5.4]
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Enable Corepack
+        run: corepack enable
+
+      - name: Setup Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node }}
+          cache: "pnpm"
+
+      - name: Install Dependencies
+        run: pnpm install
+
+      - name: Install TypeScript Version ${{ matrix.ts }}
+        run: pnpm install typescript@${{ matrix.ts }}
+
+      - name: Test Types
+        run: pnpm type-check
```

I appreciate the extra docs but I am trying to slim this kind of file down as much as possible, so there's less to get my head around when I come back to it having forgotten everything a few months later. It's a bit of a personal preference thing, but I find docs that _could_ be inferred from what they're documenting a bit distracting. So how about something like this instead - I think it should work just as well:
```suggestion
      - uses: actions/checkout@v4
      - run: corepack enable
      - run: pnpm install
      - run: pnpm install typescript@${{ matrix.ts }}
      - run: pnpm type-check
```

I think each step is sufficiently self-documenting, and it makes it easier to see what the purpose of this job is at a glance.

Also, for the sake of consistency with the other job.

---

Yes sirrr!